src/java.base/share/classes/com/sun/crypto/provider/PBEKeyFactory.java

```diff
@@ -56,7 +56,7 @@ private PBEKeyFactory(String keytype) {
     }
 
     static {
-        validTypes = HashSet.newHashSet(17);
+        validTypes = HashSet.newHashSet(21);
```

Comment:
I think there could be sight timing variations between `HashSet` and whatever Set implementation is returned by `Set.of()` but creation only happens once as you stated, and iterations over 20-ish elements just seems like the variance would be negligible in any real-world use cases.

Reply:
How about changing to use the `HashSet(Collection<? extends E> c)` constructor? First construct a collection with the various algorithms and then construct `validType` w/ this collection. `validType` can be made unmodifiable also since it's read only.